### DISCUSSION TITLE
Recover Terminal

### DISCUSSION THREAD

**Discussion by kaaax0815** [coder/code-server#2580] [Ideas] [answered]
I want to use the Terminal as a Server for e.g. my discord bot.
If I reload the page, the bot is still running, but i don't see the console anymore, cant stop it or see the logs.
Maybe it would be a cool idea to see the terminal session even after leaving the page

**Comment by abhay-ranawat** [comment, marked as answer]
Hey you can use screen , so you can connect to it even after leaving the page

**Reply by kaaax0815** [reply]
What do you mean with screen? 
1. Do you mean tty1, tty2, ...? <- Doesnt work because i am using ssh
2. Do you mean [this screen](https://linuxize.com/post/how-to-use-linux-screen/)? <- This would make sense

**Reply by abhay-ranawat** [reply]
Yep, I meant the second one , it also has many alternatives if you want

**Reply by code-asher** [reply]
Yup something like screen or tmux is the best way to do this. We have an issue for implementing persistent terminal sessions: https://github.com/cdr/code-server/issues/2276

**Comment by sr229** [comment]
This is not the intended use of code-server, however, the said such process should terminate since the interactive session is disposed.


To reiterate: **Do not use the VSCode terminal for non-interactive tasks.**

**Reply by nooobcoder** [reply]
It does not terminate, on windo reload, it keeps running in the background until the code-server container (in case of docker) is rebooted. **It would be better if code-server kills the process when it gets disconnected from the client.**
